core/bindings.rs

```diff
@@ -137,14 +128,13 @@ pub(crate) fn initialize_context<'s>(
     codegen,
     "Deno.__op__ = function(opFns, callConsole, console) {{"
   );
-  if bindings_mode == BindingsMode::New {
+  if init_mode == InitMode::New {
     _ = writeln!(codegen, "Deno.__op__console(callConsole, console);");
   }
   for op_ctx in op_ctxs {
     if op_ctx.decl.enabled {
       // If we're loading from a snapshot, we can skip registration for most ops
-      if bindings_mode == BindingsMode::LoadedFinal
-        && !op_ctx.decl.force_registration
+      if init_mode == InitMode::FromSnapshot && !op_ctx.decl.force_registration
```

Comment:
Great that this works, but it worries me a bit that this functionality can change w/o breaking a test.

Reply:
It's because all of the extensions with ops which are added between the runtime snapshot and the cli snapshot call `force_op_registration()`. So requiring `&& !will_snapshot` to short-circuit doesn't do anything.

Here's the change in behaviour: If an embedder is using the runtime snapshot, adding their own extensions with ops _without_ calling `force_op_registration()`, creating their own snapshot on top, now this person's ops won't be registered.

Previously this wouldn't happen, but it _would_ happen if the embedder wasn't creating their own snapshot. Now there's parity in those two cases.

Reply:
Thanks for the explanation - I'm OK with that change but because it might affect embedders, we should have @bartlomieju sign off on it.

I would like to figure out how we can avoid `force_op_registration` entirely so this is probably a good motivator to do that. Ideally we should determine which extensions are added from the original snapshot and those should be force-registered. We most likely have enough information for that.

Rest of the PR LGTM, thanks!

Reply:
> Ideally we should determine which extensions are added from the original snapshot and those should be force-registered.

We essentially have that via the idempotency of `Deno.__op__registerOp()`. We had this short-circuit from when we were using the v8 api and had to create a v8 string for each op name etc. With the codegen method, do we need it? I think we can just get rid of that `continue` block.

Reply:
Simplifying this would be great. I'm in favour of that, however if we do this we definitely need to ensure that we have a test for it (we may already have one that I am unaware of) and make sure we don't regress startup time (via hyperfine benchmark).

Reply:
I wouldn't know what to test for because we're getting rid of a shortcut in this case.

Reply:
Good point. Ensuring no startup regression (on running an empty script) should be sufficient.

Reply:
Looks fine
```
Benchmark 1: deno run temp.js
  Time (mean ± σ):      28.9 ms ±   1.1 ms    [User: 1.9 ms, System: 3.3 ms]
  Range (min … max):    27.4 ms …  34.0 ms    97 runs

Benchmark 2: target/release/deno run temp.js
  Time (mean ± σ):      29.0 ms ±   0.9 ms    [User: 3.0 ms, System: 3.6 ms]
  Range (min … max):    27.5 ms …  33.1 ms    95 runs

Summary
  'deno run temp.js' ran
    1.00 ± 0.05 times faster than 'target/release/deno run temp.js'
```